lib/acts-as-taggable-array-on/taggable.rb

```diff
@@ -2,14 +2,24 @@
 
 module ActsAsTaggableArrayOn
   module Taggable
+    class TaggableError < StandardError; end
+
+    class InvalidAllowListTypeError < TaggableError
```

Comment:
It's nice to see that your error handling, it's really useful.

I would like to consider if all of this code to raise an error is actually necessary. Maybe there is a way to simplify and improve API in here?

We basically have two additional errors right now, they are:
```
ActsAsTaggableArrayOn::Taggable::TaggableError
ActsAsTaggableArrayOn::Taggable::InvalidAllowListTypeError
```
Not really readable, though? "InvalidAllowListTypeError" reminds me of my java development experience, but we can do better ruby :P

We use only one error in our code, though. Why do we define two types of errors?

One of errors has a `.to_s` method, but second one doesn't. Should we provide a way to translate this definition? Do we really need it?

What kind of errors do ActionRecord/ActiveModel throws? Would it be better to use that instead?

Reply:
I've looked into some possible errors from ActiveRecord, and it seems reasonable to inherit from `ActiveRecord::ActiveRecordError`. But can we reuse one of errors from ActioveRecord namespace or should we define our own? I haven't found anything suitable for our cause in existing list off errors.

I would propose to use ActiveRecord namespace for our errors, e.g:
```
class ActiveRecord::<error_name> <  ActiveRecord::ActiveRecordError; end
```
Regarding a name, I propose something simpler, maybe `TagDefinitionError` or maybe "TaggableArrayError"?